### PR TITLE
XML generation within AppCast and AppCastItem

### PR DESCRIPTION
**Pull request by cjmurph**
makes appcast generation easier.
change to xml.linq for parsing for ease of use and readability.
update appcastgenerator to use built in xml generation methods.

### CODE REVIEW

**Comment by Deadpikle** [comment]
Hi @cjmurph! Thanks for your continued contributions.

In general, looks good. Definitely cleans things up. A few items of note, though:

-All functions that were added need documentation. Build in Debug mode for these to pop up. This includes `AppCastItem.Parse`, all the `AppCastItem` operators, `AppCastItem.GetXElement`, and `AppCast.GenerateAppCastXml`.
-I'll make another note on another part of the code here in just a moment.

File: NetSparkle/AppCast.cs [hunk 1]
```diff
@@ -170,155 +190,175 @@ private bool ReadStream(Stream inputstream, string signature)
...
+        //    }
+
+        //    // sort versions in reverse order
+        //    Items.Sort((item1, item2) => -1 * item1.CompareTo(item2));
+        //}
+
         /// <summary>
         /// Returns sorted list of updates between current and latest. Installed is not included.
         /// </summary>
-        public List<AppCastItem> GetUpdates()
```
Comment (Deadpikle) [inline NetSparkle/AppCast.cs]: Can we change this back to a `List`, please? `GenerateAppCastXml` takes a `List`. Was there another reason I'm missing for this to change back to a `[ ]`?

File: NetSparkle/AppCast.cs [hunk 2]
```diff
@@ -170,155 +190,175 @@ private bool ReadStream(Stream inputstream, string signature)
...
+                var currentItem = AppCastItem.Parse(_config.InstalledVersion, _config.ApplicationName, _castUrl, item, _logWriter);
+                Items.Add(currentItem);
             }
 
             // sort versions in reverse order
-            _items.Sort((item1, item2) => -1 * item1.CompareTo(item2));
+            Items.Sort((item1, item2) => -1 * item1.CompareTo(item2));
         }
 
+        //private void Parse(XmlReader reader)
```
Comment (Deadpikle) [inline NetSparkle/AppCast.cs]: It's alright to just get rid of this code instead of commenting it out.

File: NetSparkle/AppCast.cs (hunk 1 above)
Comment (cjmurph) [reply NetSparkle/AppCast.cs]: Not sure what happened there, it should be List everywhere. Must have happened when I merged from upstream I think.

**Comment by Deadpikle** [comment]
Again, sorry for the wait here. My mind hasn't been on NetSparkle lately. Appreciate your contribution! All merged in. 👍

**Pull request merged**